lib/index.js

```diff
@@ -151,6 +156,39 @@ function lintFile(css, options, filename) {
   return processor.process(css, options.postcss);
 }
 
+/**
+ * Add the encapsulation plugins
+ * to an existing Array of plugins.
+ *
+ * N.B. the plugins must be inserted after
+ * postcss[-easy]-import otherwise encapsulation
+ * is applied only to the entry file selectors.
+ *
+ * @param {Array} plugins Existing Array of plugins or empty Array
+ * @returns {Array} New set of plugins
+ */
+function withEncapsulation(plugins) {
+  var encapsulation = require('./encapsulation');
+  function encapsulationInjectIndex(index, plugin, currentIndex) {
+    if (
+      plugin.postcssPlugin === 'postcss-easy-import' ||
+      plugin.postcssPlugin === 'postcss-import'
+    ) {
+      index = currentIndex + 1;
+    }
+    return index;
+  }
+
+  plugins.splice(
+    plugins.reduce(encapsulationInjectIndex, plugins.length),
+    0,
+    encapsulation.resetGeneric,
+    encapsulation.resetInherited
+  );
+
+  return plugins;
+}
```

Comment:
I think lodash's `findIndex` could help simplify this code:

``` js
var findIndex = require('lodash.findindex');

function withEncapsulation(plugins) {
  var encapsulation = require('./encapsulation');

  var injectIndex = findIndex(plugins, function(p) {
    return /postcss-(?:easy-)?import/.test(p.postcssPlugin);
  }) + 1;

  plugins.splice(
    injectIndex,
    0,
    encapsulation.resetGeneric,
    encapsulation.resetInherited
  );

  return plugins;
}
```

One could argue that we only need to match against `postcss-easy-import` as it's part of the core preprocessor and wraps `postcss-import` anyway. If so we could simplify it further too:

``` js
var injectIndex = findIndex(plugins, ['postcssPlugin', 'postcss-easy-import']) + 1;
```